File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.cpp
class Solution {
public:
    bool yesno(string s){
        string temp=s;
        reverse(s.begin(), s.end());
        return s == temp;
    }
    string firstPalindrome(vector<string>& words) {
        for (auto str: words){
            if (yesno(str))
                return str;
        }
        return "";
    }
};

File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.rb
# @param {String[]} words
# @return {String}
def yesno(str)
    i, j = 0, str.length() - 1
    while i < j
        if str[i] == str[j]
            i += 1
            j -= 1
        else
            return false
        end
    end
    return true
end

def first_palindrome(words)
    words.each do |word|
        if yesno(word)
            return word
        end
    end
    return ""
end